core/execution/future/market.go

```diff
@@ -5615,6 +5622,16 @@ func (m *Market) AmendAMM(ctx context.Context, amend *types.AmendAMM, determinis
 
 	m.amm.Confirm(ctx, pool)
 	m.matching.UpdateAMM(pool.AMMParty)
+	// first check if the updated pool is one-sided
+	if pool.Parameters.LowerBound == nil || pool.Parameters.UpperBound == nil {
```

Comment:
Hum, that’s a good point. We do it for normal orders I believe. Anything which could move the reference price should be covered here basically.

Reply:
I wanted to avoid excessive calls while the vAMM stuff was still a bit of an unknown in terms of performance impact, hence the explicit checks for the cases where reference prices could be added/disappeared. If there is a rebase order, and the amendment doesn't change/add/remove a side, then the existing call to check for reference moves is enough, and this call is redundant (called towards the tail end of submission of the rebased order).

Feel free to remove the conditions if my reasoning is incorrect, of course.

Reply:
I think its too complicated to try to guess, the call to `m.checkForReferenceMoves()` after the rebasing order will not include the amended AMM since we only "confirm" it afterwards so the refernce will likely move again. So the safest thing to do is to just check it always.

If performance ends up being bad then we can do things like, pass in a flag when we submit the rebasing order so that we don't call `m.checkForReferenceMoves()` there, and only call it afterwards but I'd rather only do that when we know it is a perf issue.